packages/presentation/src/PresentationTool.tsx

```diff
@@ -416,6 +428,30 @@ export default function PresentationTool(props: {
     idRef.current = params.id
   })
 
+  const workspace = useWorkspace()
+
+  const getCommentIntent = useCallback<CommentIntentGetter>(
+    ({ id, type, path }) => {
+      if (frameStateRef.current.url) {
+        return {
+          title: frameStateRef.current.title || frameStateRef.current.url,
+          name: 'edit',
+          params: {
+            id,
+            path,
+            type,
+            inspect: COMMENTS_INSPECTOR_NAME,
+            workspace: workspace.name,
+            mode: name,
+            preview: params.preview,
+          },
+        }
+      }
+      return undefined
+    },
+    [name, params, workspace.name],
```

Comment:
Is it possible to make just `params.preview` the dependency so that other changes to `params` don't create new instances of `getCommentIntent`?
```suggestion
    [name, params.preview, workspace.name],
```